# Livro para consulta:
- https://jakevdp.github.io/PythonDataScienceHandbook/03.08-aggregation-and-grouping.html
- https://jakevdp.github.io/PythonDataScienceHandbook/03.09-pivot-tables.html
    

# 1. Importando bibliotecas <a name="import"></a>

<div style="text-align: right"
     
[Voltar ao índice](#Contents)

In [1]:
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt
import datetime

%matplotlib inline

# 2. Carregando o dataframe SINASC <a name="read"></a>
<div style="text-align: right"
     
[Voltar ao índice](#Contents)

In [2]:
sinasc_raw = pd.read_csv('SINASC_RO_2019.csv')
sinasc_raw.head()

ORIGEM   CODESTAB  CODMUNNASC  LOCNASC  IDADEMAE  ESTCIVMAE  \
0       1  2679477.0      110001        1        19        5.0   
1       1  2679477.0      110001        1        29        2.0   
2       1  2679477.0      110001        1        37        9.0   
3       1  2516500.0      110001        1        30        5.0   
4       1  2516500.0      110001        1        30        2.0   

            ESCMAE  CODOCUPMAE  QTDFILVIVO  QTDFILMORT  ...  KOTELCHUCK  \
0      8 a 11 anos         NaN         0.0         0.0  ...           5   
1      8 a 11 anos    999992.0         1.0         0.0  ...           5   
2      8 a 11 anos    513205.0         2.0         0.0  ...           5   
3  12 anos ou mais    231205.0         0.0         0.0  ...           4   
4      8 a 11 anos    999992.0         1.0         0.0  ...           5   

  CONTADOR munResStatus munResTipo               munResNome  munResUf  \
0        1        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
1        2        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
2        3        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   
3        4        ATIVO      MUNIC  Alto Alegre dos Parecis  Rondônia   
4        5        ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia   

   munResLat munResLon  munResAlt  munResArea  
0  -11.93554 -61.99982      338.0    7067.025  
1  -11.93554 -61.99982      338.0    7067.025  
2  -11.93554 -61.99982      338.0    7067.025  
3  -12.13178 -61.85308      397.0    3958.273  
4  -11.93554 -61.99982      338.0    7067.025  

[5 rows x 69 columns]

# Tarefa 1

### 1. Idade media das mães e dos pais por município (coluna munResNome)


In [3]:
sinasc_1 = sinasc_raw[['IDADEPAI', 'IDADEMAE','munResNome']]
sinasc_1

IDADEPAI  IDADEMAE               munResNome
0          26.0        19    Alta Floresta D'Oeste
1          24.0        29    Alta Floresta D'Oeste
2          32.0        37    Alta Floresta D'Oeste
3          24.0        30  Alto Alegre dos Parecis
4          27.0        30    Alta Floresta D'Oeste
...         ...       ...                      ...
27023      35.0        32                  Vilhena
27024       NaN        19              Chupinguaia
27025      30.0        24                  Vilhena
27026      24.0        21                  Vilhena
27027       NaN        18  Novo Horizonte do Oeste

[27028 rows x 3 columns]

In [4]:
sinasc_1.groupby('munResNome').mean()

IDADEPAI   IDADEMAE
munResNome                                     
Alta Floresta D'Oeste      29.548701  25.991826
Alto Alegre dos Parecis    29.179104  24.844156
Alto Paraíso               28.833333  24.960177
Alvorada D'Oeste           30.757282  25.770732
Ariquemes                  32.466667  25.607866
Buritis                    30.812500  25.559177
Cabixi                     34.333333  26.037500
Cacaulândia                36.333333  25.546667
Cacoal                     30.931854  26.890102
Campo Novo de Rondônia     30.500000  24.825301
Candeias do Jamari         29.894737  25.243323
Castanheiras               30.392857  27.281250
Cerejeiras                 31.652778  27.240506
Chupinguaia                29.629630  25.333333
Colorado do Oeste          34.354839  27.602804
Corumbiara                 32.470588  24.716535
Costa Marques              30.613636  24.412017
Cujubim                    31.400000  24.482927
Espigão D'Oeste            31.198113  26.093333
Governador Jorge Teixeira  29.222222  24.401786
Guajará-Mirim              30.470588  25.462228
Itapuã do Oeste            42.500000  26.064220
Jaru                       32.197183  25.982974
Ji-Paraná                  31.132239  26.361137
Machadinho D'Oeste         31.326923  25.480000
Ministro Andreazza         33.849315  27.904762
Mirante da Serra           32.154762  26.304878
Monte Negro                34.678571  25.648241
Município ignorado - RO    22.000000  24.000000
Nova Brasilândia D'Oeste   30.520833  26.222222
Nova Mamoré                31.692308  25.326389
Nova União                 31.371429  26.894231
Novo Horizonte do Oeste    31.588235  26.938776
Ouro Preto do Oeste        31.233244  26.375594
Parecis                    32.473684  26.454545
Pimenta Bueno              29.745174  25.199675
Pimenteiras do Oeste       35.600000  25.075000
Porto Velho                32.352679  26.312670
Presidente Médici          30.150754  25.796053
Primavera de Rondônia      31.888889  27.023256
Rio Crespo                 28.000000  26.660000
Rolim de Moura             30.360479  26.600863
Santa Luzia D'Oeste        30.703297  26.121212
Seringueiras               30.500000  24.910714
São Felipe D'Oeste         31.395833  26.481481
São Francisco do Guaporé   31.638889  25.660066
São Miguel do Guaporé      32.666667  25.926346
Teixeirópolis              31.500000  26.031250
Theobroma                  30.083333  25.536082
Urupá                      31.008929  26.688235
Vale do Anari              34.000000  26.096774
Vale do Paraíso            29.976744  25.679012
Vilhena                    34.152941  26.433962

### 2. Peso médio dos bebes por sexo que nasceram no dia do seu aniversário por faixas de escolaridade mae
Ex: Você, aluna(o), nasceu no dia 10/01, então você precisa filtrar o conjunto de dados nessa data e calcular o peso médio dos bebês de cada sexo por faixa de escolaridade da mãe.

In [5]:
sinasc_raw[sinasc_raw['DTNASC'] == '2019-09-09'].groupby(['ESCMAE', 'SEXO'])['PESO'].mean()

ESCMAE           SEXO     
1 a 3 anos       Feminino     3780.000000
12 anos ou mais  Feminino     3189.266667
                 Masculino    3212.421053
4 a 7 anos       Feminino     3136.666667
                 Masculino    3343.200000
8 a 11 anos      Feminino     3267.476190
                 Masculino    3330.695652
Name: PESO, dtype: float64

### 3. Qual o municipio que nasceu menos bebe em 2019?
    - qual a idade media, maxima, minima das maes nesse municipio?
    - qual a idade media, maxima, minima dos pais nesse municipio?

In [6]:
sinasc_1.munResNome.value_counts().tail(2)

Castanheiras               32
Município ignorado - RO     1
Name: munResNome, dtype: int64

In [7]:
sinasc_1[sinasc_1['munResNome'] == 'Castanheiras'].groupby('munResNome').aggregate(['min', np.median, max])

IDADEPAI              IDADEMAE           
                  min median   max      min median max
munResNome                                            
Castanheiras     17.0   31.5  43.0       17   27.0  39

### 4. Qual o municipio que nasceu mais bebe no mês de março?
    - qual a quantidade de filhos vivos media, maxima, minima nesse municipio?
    - qual a idade media, maxima, minima dos pais nesse municipio?



In [20]:
sinasc_raw['MES'] = pd.DatetimeIndex(sinasc_raw['DTNASC']).month

In [19]:
sinasc_2 = sinasc_raw[sinasc_raw['MES']==3]
sinasc_2.munResNome.value_counts().head(2)

Porto Velho    744
Ji-Paraná      188
Name: munResNome, dtype: int64

In [24]:
sinasc_3 = sinasc_2[['IDADEPAI', 'QTDFILVIVO','munResNome']]
sinasc_3[sinasc_3['munResNome'] == 'Porto Velho'].groupby('munResNome').aggregate(['min', np.median, max])

IDADEPAI              QTDFILVIVO            
                 min median   max        min median  max
munResNome                                              
Porto Velho     19.0   34.0  62.0        0.0    1.0  9.0

### Analise as respostas encontradas, tire algum insight delas, conte pra gente algo encontrado nos dados. Algo que você julgue relevante e novo pra você.

A idade máxima dos pais para os municios com mais e menos nascimentos é drasticamente diferente sendo em Castanheiras a maior idade para os Pai foi de 43 enquanto na cidade de Porto Velho a maior idade foi de 62, talvez a maior amplitude de idades da cidade de Porto velho aconteça devido ao valor de 'n' pois o número de nascimento Castanheiras foi de 32 ao longo do ano apenas em Março foi 744.

In [21]:
sinasc_raw.dtypes

ORIGEM          int64
CODESTAB      float64
CODMUNNASC      int64
LOCNASC         int64
IDADEMAE        int64
               ...   
munResLat     float64
munResLon     float64
munResAlt     float64
munResArea    float64
MES             int64
Length: 70, dtype: object

In [22]:
sinasc_raw.columns

Index(['ORIGEM', 'CODESTAB', 'CODMUNNASC', 'LOCNASC', 'IDADEMAE', 'ESTCIVMAE',
       'ESCMAE', 'CODOCUPMAE', 'QTDFILVIVO', 'QTDFILMORT', 'CODMUNRES',
       'GESTACAO', 'GRAVIDEZ', 'PARTO', 'CONSULTAS', 'DTNASC', 'HORANASC',
       'SEXO', 'APGAR1', 'APGAR5', 'RACACOR', 'PESO', 'IDANOMAL', 'DTCADASTRO',
       'CODANOMAL', 'NUMEROLOTE', 'VERSAOSIST', 'DTRECEBIM', 'DIFDATA',
       'DTRECORIGA', 'NATURALMAE', 'CODMUNNATU', 'CODUFNATU', 'ESCMAE2010',
       'SERIESCMAE', 'DTNASCMAE', 'RACACORMAE', 'QTDGESTANT', 'QTDPARTNOR',
       'QTDPARTCES', 'IDADEPAI', 'DTULTMENST', 'SEMAGESTAC', 'TPMETESTIM',
       'CONSPRENAT', 'MESPRENAT', 'TPAPRESENT', 'STTRABPART', 'STCESPARTO',
       'TPNASCASSI', 'TPFUNCRESP', 'TPDOCRESP', 'DTDECLARAC', 'ESCMAEAGR1',
       'STDNEPIDEM', 'STDNNOVA', 'CODPAISRES', 'TPROBSON', 'PARIDADE',
       'KOTELCHUCK', 'CONTADOR', 'munResStatus', 'munResTipo', 'munResNome',
       'munResUf', 'munResLat', 'munResLon', 'munResAlt', 'munResArea', 'MES'],
      dtype='ob

In [11]:
sinasc_raw.select_dtypes(exclude=['int64','float64'])

ESCMAE         GESTACAO GRAVIDEZ    PARTO      DTNASC  \
0          8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-02-19   
1          8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-02-21   
2          8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-02-25   
3      12 anos ou mais  37 a 41 semanas    Única  Cesáreo  2019-03-20   
4          8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-03-23   
...                ...              ...      ...      ...         ...   
27023  12 anos ou mais  32 a 36 semanas    Única  Cesáreo  2019-12-13   
27024      8 a 11 anos  37 a 41 semanas    Única  Vaginal  2019-10-04   
27025      8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-08-02   
27026      8 a 11 anos  32 a 36 semanas    Única  Vaginal  2019-12-23   
27027      8 a 11 anos  37 a 41 semanas    Única  Cesáreo  2019-12-27   

            SEXO RACACOR  DTCADASTRO CODANOMAL VERSAOSIST   DTRECEBIM  \
0      Masculino  Branca  2019-03-08       NaN     3.2.01  2019-03-08   
1       Feminino  Branca  2019-03-08       NaN     3.2.01  2019-03-08   
2       Feminino  Branca  2019-03-08       NaN     3.2.01  2019-03-08   
3       Feminino   Parda  2019-04-03       NaN     3.2.01  2019-04-09   
4       Feminino   Parda  2019-04-03       NaN     3.2.01  2019-04-09   
...          ...     ...         ...       ...        ...         ...   
27023  Masculino  Branca  2020-09-28       NaN     3.2.01  2020-10-13   
27024  Masculino  Branca  2019-10-16       NaN     3.2.01  2019-10-21   
27025  Masculino   Parda  2019-08-07       NaN     3.2.00  2019-08-07   
27026  Masculino   Parda  2020-01-07       NaN     3.2.00  2020-01-07   
27027  Masculino   Parda  2020-10-15       NaN     3.2.01  2020-10-16   

      munResStatus munResTipo               munResNome  munResUf  
0            ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia  
1            ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia  
2            ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia  
3            ATIVO      MUNIC  Alto Alegre dos Parecis  Rondônia  
4            ATIVO      MUNIC    Alta Floresta D'Oeste  Rondônia  
...            ...        ...                      ...       ...  
27023        ATIVO      MUNIC                  Vilhena  Rondônia  
27024        ATIVO      MUNIC              Chupinguaia  Rondônia  
27025        ATIVO      MUNIC                  Vilhena  Rondônia  
27026        ATIVO      MUNIC                  Vilhena  Rondônia  
27027        ATIVO      MUNIC  Novo Horizonte do Oeste  Rondônia  

[27028 rows x 15 columns]